src/ucp/wireup/wireup_cm.c

```diff
@@ -855,7 +856,6 @@ static void ucp_cm_server_conn_notify_cb(uct_ep_h ep, void *arg,
         /* if reject is arrived on server side, then UCT does something wrong */
         ucs_assert(status != UCS_ERR_REJECTED);
         cm_lane = ucp_ep_get_cm_lane(ucp_ep);
-        ucp_ep->flags &= ~UCP_EP_FLAG_LOCAL_CONNECTED;
```

Comment:
what was the reason it was added in the first place?

Reply:
most likely copy-past programming.. reset of the flag before set_ep_failed makes sense for transport failure and after getting disconnect callback to avoid "double" disconnect

Reply:
i see other places which drop LOCAL_CONNECTED flag, should they be updated as well?

Reply:
actually it's hard to answer definitely and cover all possible cases without regular testing of link failure because one-sided disconnect with live link and transport error without possibility to get response have different protocols flow and have to be handled by user different way. IMO it's better to investigate cases one-by-one.

Reply:
i'm confused, because https://github.com/openucx/ucx/pull/5075/files#diff-36e34c1df2a6c773b9dda08dd8bf17eeR328 just added the dropping of LOCAL_CONNECTED flag before set_ep_failed, so which one it should be?
IMO we should never drop because ucp_worker_set_ep_failed() calls ucp_ep_cm_disconnect_cm_lane() if needed